e2e_test/batch/basic/unnest.slt.part

```diff
@@ -83,3 +83,38 @@ select distinct unnest(array[1,1,2,3,1]) as x;
 1
 2
 3
+
+query I
+select * from unnest(array[0,1,2]) with ordinality;
+----
+0 1
+1 2
+2 3
+
+query I
+select * from unnest(array[0,1,2]) with ordinality, unnest(array[3,4]) with ordinality as unnest_2;
+----
+0 1 3 1
+0 1 4 2
+1 2 3 1
+1 2 4 2
+2 3 3 1
+2 3 4 2
+
+statement ok
+create table t(arr varchar[]);
+
+statement ok
+insert into t values (Array['a','b', 'c']), (Array['d','e']);
+
+query I
+select * from t cross join unnest(t.arr) WITH ORDINALITY AS x(elts, num);
+----
+{a,b,c} c 3
+{a,b,c} b 2
+{a,b,c} a 1
+{d,e}   e 2
+{d,e}   d 1
```

Comment:
The order is reversed. I think this is because of hashjoin?  🤔 

Not related with with ordinality, but looks a little funny with it.
```
dev=> select * from t cross join unnest(t.arr) ;
   arr   | unnest
---------+--------
 {a,b,c} | c
 {a,b,c} | b
 {a,b,c} | a
 {d,e}   | e
 {d,e}   | d
```

Reply:
Maybe you should use `rowsort` or order by to make it deterministic. Our apply doesn't preserve the original order, because after unnesting, it would become a hash join.